Sources/Core/Microsoft.StreamProcessing/Aggregates/TopKAggregate.cs

```diff
@@ -6,27 +6,37 @@
 using System.Collections.Generic;
 using System.Diagnostics.Contracts;
 using System.Linq.Expressions;
+using Microsoft.StreamProcessing.Internal;
 
 namespace Microsoft.StreamProcessing.Aggregates
 {
-    internal sealed class TopKAggregate<T> : SortedMultisetAggregateBase<T, List<RankedEvent<T>>>
+    internal sealed class TopKAggregate<T> : ISummableAggregate<T, ITopKState<T>, List<RankedEvent<T>>>
     {
         private readonly Comparison<T> compiledRankComparer;
         private readonly int k;
 
-        public TopKAggregate(int k, QueryContainer container) : this(k, ComparerExpression<T>.Default, container) { }
+        public TopKAggregate(int k, IComparerExpression<T> rankComparer, QueryContainer container, bool isHoppingWindow)
+            : this(k, rankComparer, ComparerExpression<T>.Default, container, isHoppingWindow) { }
 
-        public TopKAggregate(int k, IComparerExpression<T> rankComparer, QueryContainer container)
-            : this(k, rankComparer, ComparerExpression<T>.Default, container) { }
-
-        public TopKAggregate(int k, IComparerExpression<T> rankComparer, IComparerExpression<T> overallComparer, QueryContainer container)
-            : base(ThenOrderBy(Reverse(rankComparer), overallComparer), container)
+        public TopKAggregate(int k, IComparerExpression<T> rankComparer, IComparerExpression<T> overallComparer,
+            QueryContainer container, bool isHoppingWindow)
         {
             Contract.Requires(rankComparer != null);
             Contract.Requires(overallComparer != null);
             Contract.Requires(k > 0);
             this.compiledRankComparer = Reverse(rankComparer).GetCompareExpr().Compile();
             this.k = k;
+
+            Expression<Func<Func<SortedDictionary<T, long>>, ITopKState<T>>> template;
+            if (isHoppingWindow)
+                template = (g) => new HoppingTopKState<T>(k, compiledRankComparer, g);
+            else
+                template = (g) => new SimpleTopKState<T>(g);
+
+            var combinedComparer = ThenOrderBy(Reverse(rankComparer), overallComparer);
+            var generator = combinedComparer.CreateSortedDictionaryGenerator<T, long>(container);
+            var replaced = template.ReplaceParametersInBody(generator);
+            initialState = Expression.Lambda<Func<ITopKState<T>>>(replaced);
```

Comment:
>initialState [](start = 12, length = 12)

Did you disable the stylecop rules? instance members should be prefixed by "this.". Please make sure to follow style rules.